src/evo/deterministicmns.cpp

```diff
@@ -906,7 +902,7 @@ void CDeterministicMNManager::DecreasePoSePenalties(CDeterministicMNList& mnList
     // only iterate and decrease for valid ones (not PoSe banned yet)
     // if a MN ever reaches the maximum, it stays in PoSe banned state until revived
     mnList.ForEachMN(true, [&](const CDeterministicMNCPtr& dmn) {
-        if (dmn->pdmnState->nPoSePenalty > 0 && dmn->pdmnState->nPoSeBanHeight == -1) {
+        if (dmn->pdmnState->nPoSePenalty > 0 && dmn->pdmnState->IsBanned()) {
```

Comment:
Hmm, I'm gonna try to create a regression test about this... I'm kind of surprised tests didn't catch this somewhere

Reply:
Nevermind, dkgerrors.py did catch it